src/Core/Domain/Category/Command/AbstractEditCategoryCommand.php

```diff
@@ -0,0 +1,317 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Category\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CategoryConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Category\ValueObject\CategoryId;
+
+/**
+ * Class EditRootCategoryCommand edits given root category.
+ */
+abstract class AbstractEditCategoryCommand
```

Comment:
seems like you had some similar ideas as I did, but @jolelievre might not be up to this hehe :smile: 
https://github.com/PrestaShop/PrestaShop/pull/30231#discussion_r1025048099

Reply:
Indeed 😅 Although I do understand the need and the motivation here, but each command is supposed to be independent so they shouldn't be coupled by extension even if it's an abstract class (although it's less a problem with an abstract class maybe).

But the idea is that modifying the parent changes both commands and both contracts, which you can't assume to be true at all time. So to avoid some unexpected modification on unrelated command it is probably safer to keep commands independent. But I'm still not fully convinced of either choices... maybe we should discuss this at the next standup?